src/wp-includes/media.php

```diff
@@ -5564,6 +5680,110 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Get loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string $tag_name The tag name.
+ * @param array[] $attr Array of the attributes for the tag.
+ * @param string $context Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	// If the lazy-loading attribute already present then don't add `fetchpriority="high"`.
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		$loading_attrs['loading'] = 'lazy';
+		return $loading_attrs;
+	}
+
+	// If a 'fetchpriority' attribute with value 'high' is already provided, do not add `loading="lazy"`.
+	if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority'] ) {
+		$loading_attrs['fetchpriority'] = 'high';
+		return $loading_attrs;
+	}
+
+	// Skip lazy-loading for the overall block template, as it is handled more granularly.
+	// The skip is also applicable for fetchpriority.
+	if ( 'template' === $context ) {
+		return $loading_attrs;
+	}
+
+	// Do not lazy-load images in the header block template part, as they are likely above the fold.
+	// For classic themes, this is handled in the condition below using the 'get_header' action.
+	$header_area = WP_TEMPLATE_PART_AREA_HEADER;
+	if ( "template_part_{$header_area}" === $context ) {
+		return wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr );
+	}
+
+	// Special handling for programmatically created image tags.
+	if ( ( 'the_post_thumbnail' === $context || 'wp_get_attachment_image' === $context ) ) {
```

Comment:
```suggestion
	if ( 'the_post_thumbnail' === $context || 'wp_get_attachment_image' === $context ) {
```
These double parentheses aren't necessary here - the function doesn't use them elsewhere either.